android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -280,8 +280,8 @@ else if (RoutingController.get().hasSavedRoute())
     // by checking all the necessary permissions
     if (TrackRecorder.nativeIsEnabled())
     {
-      TrackRecorder.nativeSetEnabled(false);
-      startTrackRecording(false);
+      if (!startTrackRecording(false))
+        TrackRecorder.nativeSetEnabled(false);
```

Comment:
With a red alert the dialog may not be necessary at all (if alert properly explains the issue and maybe even opens the battery/app system settings?)

Reply:
But that notification may become red due to many reasons other than 
battery saver issue such as if location fix is not there because user entered in tunnel or indoor (temporarily). 

If suppose battery saver is turned off and still notification became red due to some location errors.

Reply:
Can notification properly handle different cases? Then it would perfectly solve these issues.

Reply:
> Can notification properly handle different cases? Then it would perfectly solve these issues.

Yess, notification will switch to red as soon as location updates to track recorder will be blocked for 30 seconds.

Reply:
> I thought about it twice - we have a red alert in the notification bar if the location doesn't work. I think that it will be fine to show to start the recording and then show the dialog about power saving in non-blocking mode. 
> 
> 

Yess, that will also solve the issue of track deletion.

I have just one more concern, since there are two non blocking dialogs simultaneously one after other (post notifications and battery saver). Wouldn't they interfere each other? Like killing the first dialog and showing second one without even taking user input?

Reply:
> @kavikhalique do you see any issues with a test case you described?

No, it works properly now. I am trying to cover all the cases where it could fail then i will test it too. Will provide feedback in some time.